packages/pds/src/app-view/api/app/bsky/feed/getTimeline.ts

```diff
@@ -111,3 +118,56 @@ export default function (server: Server, ctx: AppContext) {
     },
   })
 }
+
+const ensureReadAfterWrite = async (
+  ctx: AppContext,
+  requester: string,
+  res: ApiRes<OutputSchema>,
+): Promise<OutputSchema> => {
+  const clock = getClock(res.headers)
+  if (!clock) return res.data
+  const localSrvc = ctx.services.local(ctx.db)
+  const local = await localSrvc.getRecordsSinceClock(requester, clock, [
+    ids.AppBskyFeedPost,
+  ])
+  const formatted = await findAndFormatPostsToInsert(
+    localSrvc,
+    local.posts,
+    res.data.feed,
+  )
+  if (formatted.length === 0) return res.data
+  const feed = [...res.data.feed]
+  for (const post of formatted) {
+    if (post === null) continue
+    let inserted = false
+    for (let i = 0; i < feed.length; i++) {
+      if (feed[i].post.indexedAt < post.indexedAt) {
+        feed.splice(i, 0, { post })
+        inserted = true
+        break
+      }
+    }
+    if (!inserted) {
+      feed.push({ post })
+    }
+  }
+  return {
+    cursor: res.data.cursor,
```

Comment:
I do like these passthroughs since they're a little more relaxed about future changes to the response lexicons:
```suggestion
    ...res.data,
```

Reply:
ah yup good call 👍